### PR TITLE
#456(issue)-Implemented wrapping primitive values for NSInvocation category resultOfInvokingOn:

### PR DESCRIPTION
**Pull request by vasilenkoigor**
This PR solving [issue](https://github.com/appsquickly/Typhoon/issues/456)
As a result, in TCFWrapValues category of NSInvocation was implemented method that returns returnValue of current invocation and wrap primitive value if needed. 
And used for `resultOfInvokingOn:`, because if function `[self getReturnValue:]` returns primitive value, id type is will be wrong. 

Tests for this case will be implemented as soon.

### CODE REVIEW

**Comment by spbvasilenko** [comment]
@alexgarbarev Please, review =)

File: Source/Factory/Internal/NSInvocation+TCFWrapValues.h [hunk 1]
```diff
@@ -17,4 +17,7 @@
 /** Get argument at index, wrapping primitive values in NSValue if needed */
 - (id)typhoon_getArgumentObjectAtIndex:(NSInteger)idx;
 
+/** Get return value, wrapping primitive values in NSValue if needed*/
+- (id)typhoon_getReturnValue;
```
Comment (alexgarbarev) [inline Source/Factory/Internal/NSInvocation+TCFWrapValues.h]: What's the use case for that? I don't think Typhoon supports definitions for primitive values. And we also can't support injecting primitive values. Can you advice where its used?

Comment (vasilenkoigor) [inline Source/Factory/Internal/NSInvocation+TCFWrapValues.h]: @alexgarbarev It used for case if we create factory definition for example from UIScreen, that has bounds (CGRect). Looks please https://github.com/appsquickly/Typhoon/pull/518 :)

Comment (alexgarbarev) [inline Source/Factory/Internal/NSInvocation+TCFWrapValues.h]: Thanks for explanation! 👍 I see..

**Pull request closed without merging**